docs/concepts/style_guide.md

```diff
@@ -0,0 +1,178 @@
+```
+
+## Structure Methods Should Use Explicit Blocks
+
+These methods aid in self-documenting your tests, including indention in the
+logs. If you don’t use explicit blocks, beaker does not know how to properly
+indent your test’s output.
+
+The most common
+[structure methods](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure)
+are
+[`#test_name`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#test_name-instance_method),
+[`#step`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#step-instance_method),
+and
+[`#teardown`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#teardown-instance_method).
+
+**Good:**
+```ruby
+step 'do this thang' do
+  on(host, "echo 'do this thang'")
+end
+```
+**Bad:**
+```ruby
+step 'do this thang'
+on(host, "echo 'do this thang'")
+```
+
+## Teardowns
+
+- Return the state of the system to the way it was prior to test execution
+- Put the teardown as early in the test as possible
+
+Teardowns must be used to return the system to the state it was in prior to the
+execution or attempted execution of the test. Beaker will gather all teardowns
+encountered throughout the execution path of the test. These teardowns will all
+be executed when the test exits, even if the test exits early due to a failure
+or error.
+
+### Place Teardowns Early
+
+Teardowns can be placed anywhere in the test file or its helpers. The preferred
+style is to have a teardown step near the beginning of the test file to show the
+reader that the system state will be restored.
+
+**Good:**
+```ruby
+test_name 'The source attribute' do
+
+  target_file_on_nix = '/tmp/source_attr_test'
+  teardown do
+    on(hosts[0], "rm #{target_file_on_nix}", :accept_all_exit_codes => true)
+  end
+  ...
+end
+```
+
+Teardowns are at the mercy of the scoping of the variables necessary to perform
+the restoration of the system. This fact means that additional teardown steps
+will need to be added within the scope necessary to do their job. Effort should
+be taken to make the teardown steps prominent and readable so that it can be
+confirmed, via the logs, that the system has been restored.
+
+Teardown steps registered outside of tests should use
+[`#step`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#step-instance_method)
+to document and log what they are doing.
+
+## Acceptable Exit Codes
+
+When using the Beaker
+[`on`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Helpers/HostHelpers#on-instance_method)
+method, the default setting is that only an exit code of 0 (zero) will not
+trigger an error. When other exit codes are acceptable, the
+`:acceptable_exit_codes` key with an array of exit codes should be passed in the
+options hash to `#on`. If 0 (zero) is the only acceptable exit code, then the
+`:acceptable_exit_codes` symbol must not be used.
+
+Each example can be considered a bad example for the situation below it:
```

Comment:
This line is not clear to me.

I would also prefer to not see the use cases lumped in a single source listing.

Alternative suggestion:

`
**Good:**
* Single 0 exit code allowed
```ruby
on(host, "rm #{file_to_rm}")
```
* Single non-0 exit code allowed
```ruby
on(host, "rm #{file_to_rm}", :acceptable_exit_codes => 1)
```
* Multiple exit codes allowed
```ruby
on(host, "rm #{file_to_rm}", :acceptable_exit_codes => [0,1])
```
* Any exit code allowed
```ruby
on(host, "rm #{file_to_rm}", :accept_all_exit_codes => true)
```
`